File: java/acm2.java
import java.util.Arrays;
import java.util.Scanner;

public class acm2 {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int N = scan.nextInt();
		int p = scan.nextInt();
		int[] times = new int[N];
		for (int i = 0; i < N; i++)
			times[i] = scan.nextInt();
		int num_ac = 0;
		int penalty_time = 0;
		int total_time = 0;
		if (times[p] > 300) {
			num_ac = 0;
			penalty_time = 0;
		} else {
			num_ac++;
			penalty_time = times[p];
			total_time = times[p];
			int _i = times[p];
			Arrays.sort(times);
			for (int i = 0; i < N; i++) {
				if (total_time + times[i] > 300)
					break;
				if (times[i] != _i) {
					num_ac++;
					total_time += times[i];
					penalty_time += total_time;
				}
			}
		}
		System.out.println(num_ac + " " + penalty_time);
		scan.close();
	}
}

File: ts/src/acm2.ts
import readline = require('readline')

function calculate () {
  const p = times[firstProblem] ?? 0
  if (p > 300) {
    return [0, 0]
  }

  let totalTime = p
  times = times.filter((time) => time !== p).sort((a, b) => a - b)
  let remainingTime = 300 - totalTime
  times = times.filter((time) => {
    if (time <= remainingTime) {
      totalTime += time
      remainingTime = 300 - totalTime
      return true
    }
    return false
  })
  times.unshift(p)
  let count = 1
  return [times.length, times.reverse().reduce((a, b) => a + count++ * b, 0)]
}

let firstProblem: number
let times: number[] = []
let lineCount = 0

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line) => {
    if (lineCount % 2 === 0) {
      const [, first] = line.split(' ').map(Number)
      firstProblem = first ?? 0
    } else {
      times.push(...line.split(' ').map(Number))
      const [problemsSolved, penaltyTime] = calculate()
      console.log(`${problemsSolved} ${penaltyTime}`)
      times = []
    }
    lineCount++
  })
  .on('close', () => {
    process.exit(0)
  })
